### PR TITLE
Features/link token to domain and refresh token

### PR DESCRIPTION
**Pull request by Vicu1**

### CODE REVIEW

File: src/app/modules/users/users.controller.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
-import { Controller, Delete, Get, Patch, Post } from '@nestjs/common';
+import {Controller, Delete, Get, Patch, Post, UseGuards} from '@nestjs/common';
 import { ApiOperation, ApiTags } from '@nestjs/swagger';
 import { UsersService } from '@/app/modules/users/users.service';
+import {AuthGuard} from '@/app/middleware/guards/auth.guard';
```
Comment (Vady4ek) [inline src/app/modules/users/users.controller.ts:4]: Format with prettier

File: src/app/modules/auth/classic-auth/classic-auth.service.ts [hunk 2]
```diff
@@ -96,34 +86,47 @@ export class ClassicAuthService {
...
+      });
+      if (!payload.provider) {
+        const existingUser = await this.classicAuthRepository.findOneByEmail(payload.email);
+
+        return this.generateToken(existingUser, request);
+      }
 
+      return this.passportJsService.getNewToken(payload, request);
+    } catch (e) {
+      throw new UnauthorizedException('Invalid refresh token');
```
Comment (Vady4ek) [inline src/app/modules/auth/classic-auth/classic-auth.service.ts:125]: Add localisation.

File: src/app/modules/auth/classic-auth/classic-auth.service.ts [hunk 3]
```diff
@@ -187,51 +190,77 @@ export class ClassicAuthService {
...
+  }
+
+  private generateResetPasswordLink(token: string) {
+    return process.env.CLASSIC_AUTH_RESET_PASSWORD_LINK.replace('{token}', token);
   }
 
-  private generateResetPasswordLink (token: string) {
-    return process.env.CLASSIC_AUTH_RESET_PASSWORD_LINK
-      .replace ('{token}', token);
+  private generateToken(existingUser: ClassicAuthEntity, request: Request) {
```
Comment (Vady4ek) [inline src/app/modules/auth/classic-auth/classic-auth.service.ts:239]: Add return type.

File: src/app/modules/auth/classic-auth/dto/classic-auth-get-token.payload.dto.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { ApiProperty } from '@nestjs/swagger';
+import {IsEnum, Length} from 'class-validator';
+import {OauthProvider} from '@/app/modules/common/enums/provider.enum';
```
Comment (Vady4ek) [inline src/app/modules/auth/classic-auth/dto/classic-auth-get-token.payload.dto.ts:3]: Format with prettier.

File: src/app/modules/auth/classic-auth/dto/classic-auth-refresh-token.payload.dto.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { ApiProperty } from '@nestjs/swagger';
+import {IsString} from 'class-validator';
+export class ClassicAuthRefreshTokenPayloadDto {
+  @ApiProperty ({ example: 'eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJwcm9wcyI6eyJhdXRoUHJvdmlkZXIiOiJDTEFTU0lDIiwiZW1haWwiOiJ2aWN1Z2FpZGVpY0BnbWFpbC5jb20iLCJuYW1lIjoiVmljdSIsImlzQWN0aXZlIjp0cnVlLCJkb21haW4iOiJsb2NhbGhvc3QifSwic3ViIjoiZjJhYWRkMGMtNWVhZC00ZjZjLWJmYjMtMDlhYzM3MWE5NTZmIiwiaWF0IjoxNzE2ODk1NTE4LCJleHAiOjE3MTY4OTkxMTh9.HKBbKfQ5iAlRuUkXYTGpfauXc_NIEYkQMVMMQUIiHZgTmJyu7vtTce5d9vjHAjmzBLmsoLxWQuMcYOMeSsYL4OuQeN1sIdpBlmGF7ljLwmLv1zAZfn2q00ZpPV--AxWoA6KbtHKC6UdQLUC9hG6U1pluAEhERz0DEgJyaxQF8j3YRn4Kga0Xzj2ImySYV3-Ec7lOU8Pk9IhkDRzxgF11i6DKzAshtG6H0oQO6ltL6-o-Un6XGY5RXlMR-9dB9uF7T7', description: 'Refresh token' })
+  @IsString()
+    refreshToken: string;
```
Comment (Vady4ek) [inline src/app/modules/auth/classic-auth/dto/classic-auth-refresh-token.payload.dto.ts:6]: Format with prettier.

File: src/app/modules/auth/passport-js/passport-js.service.ts [hunk 6]
```diff
@@ -8,6 +8,8 @@ import { TokenGeneratorService } from '@/app/modules/common/token-generator.serv
 import AppConfig from '@/config/app-config';
 import { JwtService } from '@nestjs/jwt';
 import { OauthProvider } from '@/app/modules/common/enums/provider.enum';
+import {Request} from 'express';
```
Comment (Vady4ek) [inline src/app/modules/auth/passport-js/passport-js.service.ts:11]: Format with prettier.

File: src/app/modules/auth/passport-js/passport-js.service.ts [hunk 7]
```diff
@@ -83,12 +85,16 @@ export class PassportJsService {
...
       where: { token_activation_code: code },
       relations: ['user']
     });
 
+    return await this.generateToken(existingCredentials, request);
+  }
+
+  async generateToken (existingCredentials: OauthCredentialEntity, request: Request) {
     if (!existingCredentials) {
       throw new HttpException ('Not found', 404);
```
Comment (Vady4ek) [inline src/app/modules/auth/passport-js/passport-js.service.ts:99]: Add localisation & more explicit message.

**Review by Vady4ek — CHANGES_REQUESTED** [review]

File: src/app/modules/auth/passport-js/passport-js.service.ts [hunk 8]
```diff
@@ -91,14 +94,23 @@ export class PassportJsService {
...
   }
 
-  async getTokenByCode(code: string) {
+  async getTokenByCode(code: string, hostname: string, language: Language) {
     const existingCredentials = await this.oauthCredentialRepository.findOne({
       where: { token_activation_code: code },
       relations: ['user'],
     });
 
+    return await this.generateToken(existingCredentials, hostname, language);
```
Comment (vadimposcoi) [inline src/app/modules/auth/passport-js/passport-js.service.ts]: https://archive.eslint.org/docs/rules/no-return-await

**Review by Vady4ek — APPROVED** [review]

**Pull request merged**